### PR TITLE
[REVIEW] Remove requirement for runtime type info

### PR DESCRIPTION
**Pull request by zbjornson**
This makes it possible to build with -fno-rtti. (Until very recently, cuML could be built with that flag. There's only one other use of RTTI, in RMM, and it too can be eliminated.)

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

File: cpp/include/raft/linalg/cusolver_wrappers.h [hunk 1]
```diff
@@ -358,7 +359,8 @@ inline cusolverStatus_t cusolverDnsyevdx(  // NOLINT
 template <typename T>
 cusolverStatus_t cusolverDngesvd_bufferSize(  // NOLINT
   cusolverDnHandle_t handle, int m, int n, int *lwork) {
-  if (typeid(T) == typeid(float)) {
+  if (std::is_same<std::remove_cv<std::remove_reference<T>::type>::type,
+                   float>::value) {
```
Comment (jrhemstad) [inline cpp/include/raft/linalg/cusolver_wrappers.h:363]: Bit more concise and doing the same thing. 
```suggestion
  if (std::is_same<std::decay<T>::type, float>::value) {
```

Comment (zbjornson) [reply cpp/include/raft/linalg/cusolver_wrappers.h]: Thanks, change made.

**Review by cjnolet — APPROVED** [review]
LGTM

**Comment by JohnZed** [comment]
add to whitelist

**Comment by JohnZed** [comment]
@zbjornson - CI should now automatically trigger when you submit a PR (as long as you've updated the Changelog). Thank you for all the great contributions!

**Comment by zbjornson** [comment]
Thanks @JohnZed!

**Pull request merged**